OCPBUGS#13156: Updates to the 4.10 Dynamic plugin docs

opayne1 opened the pull request:
OCPBUGS#13156: Updates to the 4.10 Dynamic plugin docs

Note to reviewer: This is mainly just a reorg of content and pulls things from the existing 4.12 docs.

Version(s):
4.10 only

Issue:
https://issues.redhat.com/browse/OCPBUGS-13156

Link to docs preview:
https://opayne1.github.io/previews/OCPBUGS-13156/web_console/dynamic-plugins.html

release notes link update: https://opayne1.github.io/previews/OCPBUGS-13156/release_notes/ocp-4-10-release-notes.html#ocp-4-10-dynamic-plugin-technology-preview

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
The updates include:

- Removal of the old template link
- Updated the "About dynamic plugins" section to match the 4.12 docs
- Removed the "Getting started" section with the "Adding a tab to pods page" example because based on our conversation with PX during 4.12 that is our workaround for being able to have a GH link in the docs.
- Included the Helm chart
- Updated the structure to follow 4.12 for deploying plugin on cluster
- Updated the Additional resources section to remove old links and include the Helm chart docs

jhadvig commented on modules/about-dynamic-plug-ins.adoc:
```diff
@@ -6,29 +6,41 @@
-* _Do not_ use console CSS classes in your markup or override console CSS classes. These are not APIs and are subject to change. Using them might break your plug-in. Avoid selectors like element selectors that could affect markup outside of your plug-in’s components.
+* Follow link:https://www.i18next.com/[react-i18next] localization guidelines when creating your plugin. You can use the `useTranslation` hook like the one in the following example:
++
+[source,ymal]
```
```suggestion
[source,yaml]
```

jhadvig replied:
Although this snipped does not contain any yaml, just code snippet

opayne1 replied:
I changed to `[source,text]` to make more generic

jhadvig commented on modules/about-dynamic-plug-ins.adoc:
```diff
@@ -6,29 +6,41 @@
++
+[source,ymal]
+----
+conster Header: React.FC = () => {
```
```suggestion
const Header: React.FC = () => {
```

jhadvig commented on modules/about-dynamic-plug-ins.adoc:
```diff
@@ -6,29 +6,41 @@
+};
+----
+
+* Avoid selectors that could affect markup outside of your plugins components, such as element selectors. These are not APIs and are subject to change. Using them might break your plugin. Avoid selectors like element selectors that could affect markup outside of your plugins components.
```
```suggestion
* Avoid selectors that could affect markup outside of your plugin's components, such as element selectors. These are not APIs and are subject to change. Using them might break your plugin. Avoid selectors like element selectors that could affect markup outside of your plugins components.
```

Not a native speaker but does not look right to me

opayne1 replied:
I agree there should be an apostrophe there. Fixed

jhadvig commented on modules/deployment-plug-in-cluster.adoc:
```diff
@@ -10,29 +10,86 @@ After pushing an image with your changes to a registry, you can deploy the plugi
 
```
Should we also change the name of the file to `deployment-plugin-cluster.adoc`

opayne1 replied:
updated and also updated the others as well. Thanks!

opayne1 commented:
Hi @yapei could you help with a QE review of this PR? I think for the most part it is just reorging the 4.10 docs. I also removed the old template link and replaced with the example we did in 4.12. Thank you! 😄

yapei commented on modules/dynamic-plugins-get-started.adoc:
```diff
@@ -0,0 +1,108 @@
+}
+----
+
+.Verification
```
Shall we also mention that user should deploy && enable the plugin before visiting a Pod page to view the added tab?

opayne1 replied:
Yes, thank you! I have added the step with a reference to the doc for further reading.

yapei commented:
looks good to me, thanks!

ousleyp reviewed (commented):
Hey Olivia! I have some comments and suggestions for you to consider. I know that you are adding content to 4.10 that already exists in other versions, so it might not make sense to fix some of it in this PR. However, I think it would be worth opening one or more follow-up PRs to fix these issues in any versions that they impact.

Please feel free to ping me on Slack if you have any questions about my comments. Thanks!!
